src/components/Register/Register.js

```diff
@@ -0,0 +1,51 @@
+import './Register.css';
+import React from 'react';
+import useFormWithValidation from '../../utils/formValidationHook';
+
+function Register({setSignup, showSuccess, registerUser, setShowSuccess}) {
+
+
+const { values, handleChange, errors, isValid, resetForm } = useFormWithValidation();
+
+const handleRegistration = (event) => {
+    event.preventDefault();
+    //registering the user in the backend
+    const { email, password, name } = values;
+    registerUser(email, password, name);
+}
+
+
+if (showSuccess) {
+    return (
+        <>
+        <h3 className="popup__title">Registration successfully completed!</h3>
+        <p className="popup__text"><span onClick={() => { setSignup(false); setShowSuccess(false)}}>Sign in</span></p>
+        </>  
+    )
+}
+
+if (!showSuccess) {
+    return (
+        <> 
+        <h3 className="popup__title">Sign up</h3>
+        <form onSubmit={handleRegistration} className="popup__form">
+            <label className="popup__label" htmlFor="email">Email</label>
+            <input required className="popup__input" type="email" name="email" id="email" placeholder="Enter email" onChange={handleChange} value={values.email}/>
```

Comment:
`value={values.email}`

* There is a warning in the console. See here http://joxi.ru/BA0YjEvS1QBMJr 

React is now complaining in the console about `undefined` values. So, to make it calm down you need to add this check:  `value={values.email || ''}`


**please, make such checks in every `value` of the  inputs**